app/controllers/appointments_controller.rb

```diff
@@ -0,0 +1,75 @@
+class AppointmentsController < LoginController
+  before_action :confirm
+  before_action :set_appointment, only: %w[ show edit update destroy ]
+
+  # GET /appointments
+  # GET /appointments.json
+  def index
+    @appointments = Appointment.all
+  end
+
+  # GET /appointments/1
+  # GET /appointments/1.json
+  def show
+  end
+
+  # GET /appointments/new
+  def new
+    @appointment = Appointment.new
+  end
+
+  # GET /appointments/1/edit
+  def edit
+  end
+
+  # POST /appointments
+  # POST /appointments.json
+  def create
+    @appointment = Appointment.new(appointment_params)
+
+    respond_to do |format|
+      if @appointment.save
+        format.html { redirect_to @appointment, notice: "Appointment was successfully created." }
+        format.json { render :show, status: :created, location: @appointment }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @appointment.errors, status: :unprocessable_entity }
```

Comment:
I think the Appointment model needs some methods for @appointment.errors to work properly, since it doesn't work by default. check out https://api.rubyonrails.org/classes/ActiveModel/Errors.html

Reply:
deleted!